hubble/hubble.go

```diff
@@ -60,18 +60,18 @@ type k8sHubbleImplementation interface {
 	GetDaemonSet(ctx context.Context, namespace, name string, opts metav1.GetOptions) (*appsv1.DaemonSet, error)
 	CiliumStatus(ctx context.Context, namespace, pod string) (*models.StatusResponse, error)
 	ListCiliumEndpoints(ctx context.Context, namespace string, opts metav1.ListOptions) (*ciliumv2.CiliumEndpointList, error)
-	GetRunningCiliumVersion(ctx context.Context, namespace string) (string, error)
 	GetServerVersion() (*semver.Version, error)
+	GetHelmState(ctx context.Context, namespace string, secretName string) (*helm.State, error)
 }
 
 type K8sHubble struct {
-	client              k8sHubbleImplementation
-	params              Parameters
-	certManager         *certs.CertManager
-	ciliumVersion       string
-	manifests           map[string]string
-	semVerCiliumVersion semver.Version
-	helmYAMLValues      string
+	ctx            context.Context
```

Comment:
I think we shouldn't include the context inside the struct but rather pass it to the functions that need it. See https://pkg.go.dev/context which states:

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it.

Also see https://go.dev/blog/context-and-structs